File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        int n = strs.size();
        unordered_map<string, vector<string>> hashMap;
        for (auto &str: strs) {
            string sortedStr = str;
            sort(sortedStr.begin(), sortedStr.end());
            if (hashMap.find(sortedStr) != hashMap.end()) {
                hashMap[sortedStr].push_back(str);
            }
            else {
                hashMap[sortedStr] = {str};
            }
        }
        vector<vector<string>> result;
        for (auto &it: hashMap) {
            result.push_back(it.second);
        }
        return result;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let table = {};
    for (let i = 0; i < strs.length; i++) {
        let arrStr = strs[i].split('');
        let key = arrStr.sort();
        if (!table[key]) {
            table[key] = [];
        }
        table[key].push(strs[i]);
    }
    let keys = Object.keys(table);
    return keys.map(e => table[e]);
};